lib/telepoison.ex

```diff
@@ -46,13 +48,29 @@ defmodule Telepoison do
   def setup(opts \\ []) do
     Agent.start_link(
       fn ->
-        case Keyword.get(opts, :infer_route) do
-          nil ->
-            {:ok, &Telepoison.URI.infer_route_from_request/1}
+        infer_fn =
+          case Keyword.get(opts, :infer_route) do
+            nil ->
+              &Telepoison.URI.infer_route_from_request/1
+
+            infer_fn when is_function(infer_fn, 1) ->
+              infer_fn
+          end
 
-          infer_fn when is_function(infer_fn, 1) ->
-            {:ok, infer_fn}
-        end
+        defaults =
+          @default_attributes
```

Comment:
Instead of maintaining a `@default_attributes` list, we could explicitly ask for an `:attributes` list. This could make it so that the user is free to pass whatever they want, as it is for the `request` call

Reply:
Done!